### DISCUSSION TITLE
How is ir_drop and ir_drop_g_ratio (in IOParameters) utilized?

### DISCUSSION THREAD

**Discussion by nano-matte** [IBM/aihwkit#461] [Q&A]
Obviously the physical lead resistance as well as the conductance of the RPUs will matter for scaling of arrays due to voltage drop along the rows. Even though the RPU weights are normalized, in IOParameters there are two parameters that may be used to model the impact of actual absolute RPU conductance level (ir_drop and ir_drop_g_ratio), but from the documentation I do not understand how these are actually used by aihwkit. 

1. Could someone with insight into this give a short explanation to what "scale" ir_drop is supposed to define? Is it a fractional value of voltage drop between each cell or something else? In any case it would be hard to define this as a fixed value given that it must depend on the g value of the RPU's?
2. What is ir_drop_g_ratio actually used for? As far as I can see, this is the only place in which one can enter the information of an absolute conductance value for the RPUs. But how is it then used?

**Comment by maljoras** [comment]
Hi @nano-matte,
many thanks for the questions as it is indeed not explained very clearly. The analog tile matrix is maintained in normalized conductance values, where 1 typically is though of being gmax. The ir_drop_g_ratio is given in `g_wire / g_max` and is thus an unitless number.  The scale (`ir_drop`) is applied to scale the final `\Delta y_{IR-drop}` (as defined below in Eq (16)) scaling the effect of the IR drop that is then added to the overall output (before applying the ADC).  We have an upcoming paper explaining it more detail, where the relevant section reads:

![image](https://user-images.githubusercontent.com/17587387/216184192-f2d46e4d-dfc7-463d-bbed-427d7c5176a0.png)
![image](https://user-images.githubusercontent.com/17587387/216184019-164ad4ee-3872-40bb-a8ca-e13bce917ecd.png)

**Reply by maljoras** [reply]
Note that most other parameters (e.g. weight noise etc. ) are in relation to the normalized weights where a value of 1 corresponds to gmax.

**Reply by SyJin-pro** [reply]
Is this article published? What is the name of the article?

**Reply by maljoras-sony** [reply]
It published now, see https://www.nature.com/articles/s41467-023-40770-4
